### PR TITLE
add angsd contamination

### LINKED ISSUE #841: new module: angsd/contamination
I think it would be good to have a module for [angsd](https://bioconda.github.io/recipes/angsd/README.html)/contamination

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [ ] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jbv2**
This PR adds the angsd/contamination module.

## PR checklist

Closes #841  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/nf-core/angsd/contamination/main.nf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+workflow test_angsd_contamination {
+    
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam_bai'], checkIfExists: true),
+        []
+    ]
+
+    hapmap_file = file("https://github.com/nf-core/eager/raw/master/assets/angsd_resources/HapMapChrX.gz")
```
Comment (TCLamnidis) [inline tests/modules/nf-core/angsd/contamination/main.nf:17]:
```suggestion
    hapmap_file = [ [id:'test2'], file("https://github.com/ANGSD/angsd/raw/0.940/RES/HapMapChrX.gz")]
```
Applied the suggestion to add a meta by @nvnieuwk, and additionally changed the URL for the file to the ANGSD github page from the eager one. :)

Comment (nvnieuwk) [reply tests/modules/nf-core/angsd/contamination/main.nf]: Can these files be added to the modules branch in the test-datasets repo? Or is it a huge file?

File: modules/nf-core/angsd/contamination/main.nf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    label 'process_low'
+
+    conda "bioconda::angsd=0.940"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/angsd:0.939--h468462d_0':
+        'quay.io/biocontainers/angsd:0.939--h468462d_0' }"
+
+    input:
+    tuple val(meta), path(icounts)
+    tuple val(meta2), path hapmap_file
```
Comment (TCLamnidis) [inline modules/nf-core/angsd/contamination/main.nf:12]:
```suggestion
    tuple val(meta2), path(hapmap_file)
```

File: tests/modules/nf-core/angsd/contamination/main.nf (hunk 1 above)
Comment (TCLamnidis) [reply tests/modules/nf-core/angsd/contamination/main.nf]: That would be the cleanest solution for sure. Will do that before merging this in for sure.

Comment (nvnieuwk) [reply tests/modules/nf-core/angsd/contamination/main.nf]: Awesome!

**Review by nvnieuwk — APPROVED** [review]

**Comment by TCLamnidis** [comment]
Outstanding tasks:
-[ ] Add HapMapFile needed for tests to nf-core/test-datasets repo instead of pulling from ANGSD repo.

File: tests/modules/nf-core/angsd/contamination/main.nf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+workflow test_angsd_contamination {
+    
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_markduplicates_sorted_bam_bai'], checkIfExists: true),
+        []
+    ]
+
+    hapmap_file = [ [id:'test2'], file("https://github.com/ANGSD/angsd/raw/0.940/RES/HapMapChrX.gz")]
```
Comment (jbv2) [inline tests/modules/nf-core/angsd/contamination/main.nf:17]:
```suggestion
    hapmap_file = [ [id:'test2'], file("https://github.com/jbv2/nf-core-test-datasets/raw/modules/data/delete_me/angsd/HapMapChrX.gz")]
```

**Review by jfy133 — APPROVED** [review]
HapMap was already on nf-core testdatasets @TCLamnidis ;)

**Pull request merged**